tags.go

```diff
@@ -200,3 +206,66 @@ func (client *Client) DeleteTag(id ID) error {
 	err := client.Mutate(&m, v, WithName("TagDelete"))
 	return HandleErrors(err, m.Payload.Errors)
 }
+
+// ReconcileTags manages tags API operations for TaggableResourceInterface implementations
+//
+// Tags not in 'tagsWanted' will be deleted, new tags from 'tagsWanted' will be created. Reconciled tags are returned.
+func (client *Client) ReconcileTags(resourceType TaggableResourceInterface, tagsWanted []string) ([]Tag, error) {
+	var err error
+	var tagConnection *TagConnection
+	var assignedTags []Tag
+
+	// Get underlying resource type for type checking
+	switch resource := resourceType.(type) {
+	case *Domain:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *InfrastructureResource:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Repository:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Service:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *System:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *Team:
+		tagConnection, err = resource.GetTags(client, nil)
+	case *User:
+		tagConnection, err = resource.GetTags(client, nil)
+	default:
```

Comment:
I don't think that this `switch` is necessary since the `resourceType` is guaranteed to be of the `TaggableResourceInterface` (therefore the default case will never be hit).

Reply:
I agree that this would probably never happen. This `default` case was added mostly as extra guardrails for really odd edge cases

Reply:
@taimoorgit `(therefore the default case will never be hit)` is a VERY bad assumption to make.  In the future if we add a new type that flows through this codepath without the default case we'd could get unexpected behaviour or worse a crash.  You should ALWAYS handle the default case when you make a `switch` no matter if you think it will never be hit because you don't know what the future holds.

Reply:
Now that being side - @davidbloss since you are taking in a interface and the expected func on that interface is `GetTags` then yes this switch statement is wholly unnecessary.  Thats the purpose of having the function signature take in an interface.